js/ad_manager_vast.js

```diff
@@ -838,6 +847,34 @@ OO.Ads.manager(function(_, $) {
     }, this);
 
     /**
+     * Registered as a callback with the AMC, which gets called by the Ad Manager Controller when the the play head updates
+     * during ad playback.
+     * @public
+     * @method Vast#onAdPlayheadTimeChanged
+     * @param {string} eventname The name of the event for which this callback is called
+     * @param {number} playhead Current video time (seconds)
+     * @param {number} duration Duration of the current video (seconds)
+     */
+    this.onAdPlayheadTimeChanged = function(eventName, playhead, duration) {
+      var firstQuartileTime = duration / 4;
+      var midpointTime = duration / 2;
+      var thirdQuartileTime = (3 * duration) / 4;
+
+      if (!trackingEventQuartiles.firstQuartile && playhead >= firstQuartileTime) {
+        _handleTrackingUrls(currentAd, ["firstQuartile"]);
+        trackingEventQuartiles.firstQuartile = true;
+      }
+      else if (!trackingEventQuartiles.midpoint && playhead >= midpointTime) {
+        _handleTrackingUrls(currentAd, ["midpoint"]);
+        trackingEventQuartiles.midpoint = true;
+      }
+      else if (!trackingEventQuartiles.thirdQuartile && playhead >= thirdQuartileTime) {
```

Comment:
Are ads seekable? If we start near time 0, and seek past the 3rd quartile, should the first quartile and midpoint tracking urls also get triggered? If they should, we'll need to remove the else ifs.